source/Server/WorkItems/WorkItemLinkMapper.cs

```diff
@@ -48,28 +47,37 @@ public IResultFromExtension<WorkItemLink[]> Map(OctopusBuildInformation buildInf
             var releaseNotePrefix = store.GetReleaseNotePrefix();
             var workItemIds = commentParser.ParseWorkItemIds(buildInformation).Distinct().ToArray();
             if (workItemIds.Length == 0)
-                return ResultFromExtension<WorkItemLink[]>.Success(new WorkItemLink[0]);
+                return ResultFromExtension<WorkItemLink[]>.Success(Array.Empty<WorkItemLink>());
 
-            return ResultFromExtension<WorkItemLink[]>.Success(ConvertWorkItemLinks(workItemIds, releaseNotePrefix, baseUrl));
+            var (isSuccess, errorMessage) = TryConvertWorkItemLinks(workItemIds, releaseNotePrefix, baseUrl, out var links);
+            return isSuccess ? ResultFromExtension<WorkItemLink[]>.Success(links) : ResultFromExtension<WorkItemLink[]>.Failed(errorMessage);
         }
 
-        private WorkItemLink[] ConvertWorkItemLinks(string[] workItemIds, string? releaseNotePrefix, string baseUrl)
+        private (bool isSuccess, string? errorMessage) TryConvertWorkItemLinks(string[] workItemIds, string? releaseNotePrefix, string baseUrl, out WorkItemLink[] links)
```

Comment:
this is another place where the success/failure subtypes make more sense to me. I do not think we should be adding `out` parameters. The `return (true, null)` lines in the code had me scratching me head for a bit until I realised the out had been added, it makes the code harder to follow.